### PR TITLE
handle dependencies reported by postcss

### PR DESCRIPTION
**Pull request by sokra**
pass the dependencies reported by postcss to turbo-tasks-fs. We don't have a good mechanism to track dependencies back-timed, so we start watching some as soon we know about the dependencies (after postcss compilation)

tailwind reports dependencies as directory + glob, so our globbing need some improvements to handle these globs reported by tailwind.

e.g. tailwind will rerun when any file changed from the tailwind config `content` globs.

fixes WEB-356

### CODE REVIEW

File: crates/turbopack-node/src/transforms/postcss.rs [hunk 1]
```diff
@@ -186,12 +219,39 @@ impl Asset for PostCssTransformedAsset {
         )
         .await?;
         let JavaScriptValue::Value(val) = &*config_value else {
-            return Ok(source_content);
+            // An error happened, which has already converted into an issue.
```
Comment (alexkirsz) [inline crates/turbopack-node/src/transforms/postcss.rs:222]:
```suggestion
            // An error happened, which has already been converted into an issue.
```

File: crates/turbo-tasks-fs/src/glob.rs [hunk 2]
```diff
@@ -47,73 +49,116 @@ pub struct Glob {
...
-                i -= 1;
-            }
-        }
-    }
+                // end of expression, matched successfully
 
-    pub fn parse(input: &str) -> Result<Glob> {
-        let mut current = input;
-        let mut expression = Vec::new();
+                // backtrace for the next iteration
```
Comment (alexkirsz) [inline crates/turbo-tasks-fs/src/glob.rs:134]:
```suggestion
                // backtrack for the next iteration
```

File: crates/turbo-tasks-fs/src/glob.rs [hunk 3]
```diff
@@ -47,73 +49,116 @@ pub struct Glob {
...
+                        return Some(("", self.is_path_separator_equalivalent));
                     }
                 } else {
-                    if i == 0 {
+                    if self.index == 0 {
                         // failed to match
-                        return false;
+                        return None;
                     }
                     // backtrace
```
Comment (alexkirsz) [inline crates/turbo-tasks-fs/src/glob.rs:127]:
```suggestion
                    // backtrack
```

**Review by alexkirsz — APPROVED** [review]

File: crates/turbo-tasks-fs/src/glob.rs [hunk 4]
```diff
@@ -160,8 +237,10 @@ impl GlobPart {
 struct GlobPartMatchesIterator<'a> {
     path: &'a str,
     part: &'a GlobPart,
+    match_partial: bool,
     previous_part_is_path_separator_equalivalent: bool,
     index: usize,
+    glob_iterator: Option<Box<GlobMatchesIterator<'a>>>,
```
[the remark this answers is not in the document]
Comment (sokra) [reply crates/turbo-tasks-fs/src/glob.rs:243]: Makes sense, but I'll move that to a follow-up PR

**Pull request merged**